### ISSUE TITLE
Redirect loop after upgrade

### ISSUE THREAD

**Issue by wireland1992** [freescout-helpdesk/freescout#3070]
PHP version: 7.4.33
FreeScout version: Upgrading from 1.7.0 -> 1.8.78

I attempted to upgrade Freescout from 1.7.0 -> 1.8.78 in preparation for an upcoming PHP 8 upgrade to my server but keep getting stuck in a redirect loop when trying to log in after the upgrade.

I performed the upgrade by running ./tools/update.sh. Everything went well, there were no errors reported. I ran php artisan freescout:clear-cache, and reconfigured permissions to make bootstrap/, storage/, public/js/builds, and public/css/builds writable by the web server. I then tried to log in but got "Too many redirects" error from the browser. I checked logs but couldn't find anything. I have HTTPS configured  (it was working fine in 1.7.0). I thought maybe it was doing an HTTP -> HTTPS -> HTTP -> etc loop, but it appears to be doing HTTPS -> HTTPS -> etc. I tried disabling all modules and clearing the cache again, but still got the redirect loop.

Here's my Apache VirtualHost:

<VirtualHost *:80>
        ServerName support.domain.com
    Redirect 302 / https://support.domain.com/
</VirtualHost>

<VirtualHost *:443>
  ServerName support.domain.com
  DocumentRoot /var/www/support/public
  Options Indexes FollowSymLinks
  ErrorLog /var/www/support/storage/logs/web-server.log
  CustomLog /var/www/support/storage/logs/web-server_access.log combined
  <Directory /var/www/support/public>
    AllowOverride All
    Require all granted
  </Directory>

</VirtualHost>

I tried adding "RewriteBase /" to the Directory directive without success.

I have APP_FORCE_HTTPS=true in my .env file, toggling it on/off doesn't make a difference.
Setting APP_DEBUG=true also doesn't provide any useful info since it never displays.

According to web_server_access.log, it just shows requests to /login over and over with a 302 response code.

Any insight would be appreciated.

**Comment by freescout-helpdesk** [comment]
Check what's going on here: https://github.com/freescout-helpdesk/freescout/blob/dist/app/Http/Middleware/HttpsRedirect.php#L38

**Comment by freescout-helpdesk** [comment]
Which branch are you using in your GIT?

**Comment by freescout-helpdesk** [comment]
Should be fixed in the latest release.

**Comment by wireland1992** [comment]
I seem to be getting the redirect loop again. I upgraded from version 1.8.80 => 1.8.117

I performed the update via ./tools/update.sh and verified everything was successful. I also checked permissions, ./bootstrap/* and ./storage/* have Apache write permissions

PHP version 8.1.27

Apache virtual host is unchanged from the original post

Branch: dist

My APP_URL=https://support.example.com
I tried changing to APP_URL=http://support.example.com, which loaded the login page without any styles, the main page loaded over https, but styles loaded via http, which were blocked by CSP rules.

I removed APP_FORCE_HTTPS=true since it has no effect anymore.

I also manually turned off all modules by setting the active column to 0 in the database after the update in case a module might be causing it.
